src/lib/lib.c

```diff
@@ -43,6 +43,28 @@ int memset_s(void *dest, size_t dest_size, int data, size_t count)
 	return arch_memset_s(dest, dest_size, data, count);
 }
 
+void *memmove(void *dest, const void *src, size_t count)
```

Comment:
I'm assuming this allings with the POSIX version of memove ? If so, we should say so in the commit message.

Reply:
@lgirdwood Let me create separate PR to add memmove function. Actually all memmove functions in this PR can be replaced by memcpy because it is copying higher address to lower address. memcpy cannot be used for the opposite case.
Let me switch memmove function to memcpy in this PR.